collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -45,14 +47,44 @@ static public QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context
         appearance = appearance.toLowerCase(Locale.ENGLISH);
 
         QuestionWidget questionWidget;
+
+        // Custom Code
+        FormController formController = Collect.getInstance().getFormController();
+        String language = null;
         switch (fep.getControlType()) {
             case Constants.CONTROL_INPUT:
                 switch (fep.getDataType()) {
                     case Constants.DATATYPE_DATE_TIME:
-                        questionWidget = new DateTimeWidget(context, fep);
+//                        questionWidget = new DateTimeWidget(context, fep);
+                        // Custom code
+
+                        if(formController != null)
+                            language = formController.getLanguage();
+                        if(language != null) {
+                            if (language.toLowerCase().equals("amharic")) {
+                                questionWidget = new EthiopianDateTimeWidget(context, fep);
+                            } else {
+                                questionWidget = new DateTimeWidget(context, fep);
+                            }
+                        }else {
+                            questionWidget = new DateTimeWidget(context, fep);
+                        }
+
                         break;
                     case Constants.DATATYPE_DATE:
-                        questionWidget = new DateWidget(context, fep);
+//                        questionWidget = new DateWidget(context, fep);
+
+                        if(formController != null)
+                            language = formController.getLanguage();
+                        if(language != null) {
+                            if (language.toLowerCase().equals("amharic")) {
+                                questionWidget = new EthiopianDateTimeWidget(context, fep);
```

Comment:
This should be a `EthiopianDateWidget` and not an `EthiopianDateTimeWidget`. The other widgets in this `case` are likewise incorrect.